src/spacy/domain.clj

```diff
@@ -172,6 +178,23 @@
                        ::id (random-uuid))]}
       {::error ::cannot-delete-session})))
 
+(defn move-session [state current-user {:keys [id room time]}]
+  {:post [(s/valid? ::outcome %)]}
+  (let [session (find-in-schedule-by-id id state)
+        is-the-sponsor? (= current-user (::sponsor session))
+        can-move? (and session
+                       is-the-sponsor?
+                       (is-open-slot? state room time))
+        moved (assoc session ::room room ::time time)]
+    (if can-move?
+      {::event (-> state
+                   (update ::schedule (partial replace {session moved})))
+       ::facts [(assoc moved
+                       ::fact ::session-moved
```

Comment:
Do we want to add `::session-moved` to the set in `::facts` in L54

Reply:
Oh, good catch. I forgot to add it and yet the postcondition above passes.

I'm wondering if it's a sign that our specs are too loose or that we're using them incorrectly.

I'll look into it!

Reply:
→ #44